crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -503,6 +505,36 @@ impl GetBatchData for MeshPipeline {
     }
 }
 
+pub struct ShadowMeshPipeline;
+impl GetBatchData for ShadowMeshPipeline {
+    type Param = (SRes<RenderMeshInstances>, SRes<RenderLightmaps>);
+    // The material bind group ID, the mesh ID, and the lightmap ID,
+    // respectively.
+    type CompareData = (MaterialBindGroupId, AssetId<Mesh>, Option<AssetId<Image>>);
+
+    type BufferData = MeshUniform;
+
+    fn get_batch_data(
+        (mesh_instances, lightmaps): &SystemParamItem<Self::Param>,
+        entity: Entity,
+    ) -> Option<(Self::BufferData, Option<Self::CompareData>)> {
+        let mesh_instance = mesh_instances.get(&entity)?;
+        let maybe_lightmap = lightmaps.render_lightmaps.get(&entity);
```

Comment:
Would there be any issues with removing lightmaps in `CompareData `  for shadow pass?

Reply:
For standard materials, no… in general, possibly I guess? But probably not. 

If it did matter for a custom material then  the author could add some data into the material and use it to differentiate in the shadow_material_key fn. I’ll remove it from here.